### ISSUE TITLE
False errors about Parenthesis ( ) characters in x500 Addresses

### ISSUE THREAD

**Issue by HelenTreasure** [microsoft/idfix#16]
### Category
- [ ] Enhancement
- [X] Bug
- [ ] Question
- [ ] Documentation gap/issue

### Version

Please specify what version of IdFix you are using: [2.2.0.0]

### Expected Behavior
x500 addresses are not flagged up as invalid proxy addresses
We previously used version 1.11 and that version has the expected behaviour.

### Observed Behavior
x500 addresses are flagged up as invalid proxy addresses with a Character error, causing thousands of false errors.
The suggested correction by the tool is to remove Parenthesis  "(" and ")" characters.
This is incorrect and sync is unaffected.  Our valid x500 addresses contain these characters and are simply repopulated from Azure during the next sync if the address is changed or removed.

### Steps to Reproduce
Add an x500 proxy address to an AD object that contains the characters Parenthesis "(" or ")", such as the following:
x500:/o=ExchangeLabs/ou=Exchange Administrative Group (FYDIBOHF23SPDLT)/cn=Recipients/cn=f7a6b595bd894d3ca3d5f1cfae62d4fb-FirstLastna
Run a query in IDFix and review the results.

![image](https://user-images.githubusercontent.com/65400007/82046689-446f3680-96a9-11ea-8deb-ab7ddccb9032.png)

**Comment by pedro-roeseler** [comment]
Having the same problem in two of my clients.

**Comment by patrick-rodgers** [comment]
Thanks for letting us know - we have a proposed fix in and just need to do some further testing before we make the next release.

**Comment by colin-small** [comment]
What is the status of this fix? The issue is still open.

**Comment by 246fname** [comment]
Any update for when 2.2.0.1 or newer version will be published?

Thanks.

**Comment by danvailcolorado** [comment]
I am having the same exact same issues with 2.2.0.0. I am currently using Azure AD Connect to upload a subset of users into Azure and was cleaning up problems reported by a previous version of IDFix. I was down to just a handful of issues and then noticed I was using an older version of IDFix. I upgraded to 2.2.0.0 and then reran it and it added over 150 additional accounts in for needing charcter edit(needing to removing Parentheses from the x500 addreses). I was not paying attention to the subset of users and made a couple of the edits in IDFix, but then noticed that something was fighting me and changing the attribute in AD back. I did it again and confirmed the attributes in AD had a parenthesis in x500 address, confired that IDFix edit removed that parenthesis in AD attributes, and then confirmed that Azure AD Connect was syncing and re-adding the parenthesis back to the value that causes IDFix to report as a problem. The Service Sync Manager on Azure AD Connect is not reporting any issues. My version of Azure AD Connect is 1.4 and a few months old, so I am going to update it to 1.5, but assuming this is a bug with IDFix 2.2.0.0
